src/main/scala/is/hail/expr/ir/functions/ReferenceGenomeFunctions.scala

```diff
@@ -0,0 +1,186 @@
+  val tinterval = TInterval(tlocus)
+
+  def removeRegisteredFunctions(): Unit =
+    registered.foreach(IRFunctionRegistry.removeIRFunction)
+
+  def registerRGCode(
+    mname: String, args: Array[Type], rt: Type, isDeterministic: Boolean)(
+    impl: (EmitMethodBuilder, Array[Code[_]]) => Code[_]
+  ): Unit = {
+    val newName = rg.wrapFunctionName(mname)
+    registered += newName
+    registerCode(newName, args, rt, isDeterministic)(impl)
+  }
+
+  def registerRGCode(
+    mname: String, arg1: Type, rt: Type, isDeterministic: Boolean)(
+    impl: (EmitMethodBuilder, Code[_]) => Code[_]
+  ): Unit =
+    registerRGCode(mname, Array[Type](arg1), rt, isDeterministic) { case (mb, Array(a1)) => impl(mb, a1) }
+
+  def registerRGCode(
+    mname: String, arg1: Type, rt: Type)(
+    impl: (EmitMethodBuilder, Code[_]) => Code[_]
+  ): Unit = registerRGCode(mname, arg1, rt, isDeterministic = true)(impl)
+
+  def registerRGCode(
+    mname: String, arg1: Type, arg2: Type, rt: Type, isDeterministic: Boolean)(
+    impl: (EmitMethodBuilder, Code[_], Code[_]) => Code[_]
+  ): Unit =
+    registerRGCode(mname, Array[Type](arg1, arg2), rt, isDeterministic) { case (mb, Array(a1, a2)) => impl(mb, a1, a2) }
+
+  def registerRGCode(
+    mname: String, arg1: Type, arg2: Type, rt: Type)(
+    impl: (EmitMethodBuilder, Code[_], Code[_]) => Code[_]
+  ): Unit = registerRGCode(mname, arg1, arg2, rt, isDeterministic = true)(impl)
+
+  def registerRGCode(
+    mname: String, arg1: Type, arg2: Type, arg3: Type, arg4: Type, arg5: Type, rt: Type, isDeterministic: Boolean)(
+    impl: (EmitMethodBuilder, Code[_], Code[_], Code[_], Code[_], Code[_]) => Code[_]
+  ): Unit =
+    registerRGCode(mname, Array[Type](arg1, arg2, arg3, arg4, arg5), rt, isDeterministic) {
+      case (mb, Array(a1, a2, a3, a4, a5)) => impl(mb, a1, a2, a3, a4, a5)
+    }
+
+  def registerRGCode(
+    mname: String, arg1: Type, arg2: Type, arg3: Type, arg4: Type, arg5: Type, rt: Type)(
+    impl: (EmitMethodBuilder, Code[_], Code[_], Code[_], Code[_], Code[_]) => Code[_]
+  ): Unit = registerRGCode(mname, arg1, arg2, arg3, arg4, arg5, rt, isDeterministic = true)(impl)
+
+  def registerAll() {
+
+    val locusClass = Locus.getClass
+
+    registerRGCode("Locus", TString(), TLocus(rg)) {
+      case (mb, locusoff: Code[Long]) =>
+        val slocus = asm4s.coerce[String](wrapArg(mb, TString())(locusoff))
+        val locus = Code
+          .invokeScalaObject[String, RGBase, Locus](
+          locusClass, "parse", slocus, rgCode(mb))
+        emitLocus(mb, locus)
+    }
+
+    registerRGCode("Locus", TString(), TInt32(), TLocus(rg)) {
+      case (mb, contig: Code[Long], pos: Code[Int]) =>
+        val scontig = asm4s.coerce[String](wrapArg(mb, TString())(contig))
+        val locus = Code
+          .invokeScalaObject[String, Int, RGBase, Locus](
+          locusClass, "apply", scontig, pos, rgCode(mb))
+        emitLocus(mb, locus)
```

Comment:
I appreciate trying to re-use the existing code, but in this case, there doesn't seem much reason to pay an allocation. We could call directly to `rgCode(mb).invoke[String, Int, Unit]("checkLocus", scontig, pos)` and then emit a locus struct.